src/main/java/digital/slovensko/autogram/core/visualization/PDFVisualization.java

```diff
@@ -7,19 +7,22 @@
 import digital.slovensko.autogram.core.SigningJob;
 import digital.slovensko.autogram.ui.Visualizer;
 import eu.europa.esig.dss.model.CommonDocument;
+import eu.europa.esig.dss.pades.exception.InvalidPasswordException;
+import eu.europa.esig.dss.pdf.pdfbox.PdfBoxDocumentReader;
 
 public class PDFVisualization extends Visualization {
     private final CommonDocument document;
 
     public PDFVisualization(CommonDocument document, SigningJob job) {
         super(job);
         this.document = document;
+
+        this.checkPdfPasswordProtection();
```

Comment:
Asi by som preferoval, keby sa tento check robil "o poschodie vyssie". 

Tiez treba zvazit, ze kolko toto trva, aby sme nepredlzovali cas, za ktory sa zobrazi dokument (ked nie je zaheslovany).

Reply:
presunul som to

radovo su to 10-250ms , ked som skusal 50MB pdf tak to bolo 370ms, takze mi to pride ok